code/modules/spells/spell_types/shapeshift/_shape_status.dm

```diff
@@ -189,12 +189,15 @@
 	// is no longer in control of the shapeshifted mob, such as mindswapping out of a shapeshift
 	if(!QDELETED(source_spell) && source_spell.owner == owner)
 		source_spell.Grant(caster_mob)
+	if(usr?.contents)
```

Comment:
Why are we using `usr` here, wouldn't `owner` work here?

Reply:
using owner here somehow made it runtime on cockroaches, which qdel on death. Check some of my previous commits on this, I used owner originally to no avail

Reply:
See https://github.com/tgstation/tgstation/pull/81528/commits/c907d2a36c3dbc870fe6462aec148eba605bf067

Reply:
Does owner?.contents not work?

Reply:
the reason why i do not like `usr` here at all is because there can be rare conditions where `usr` is NOT `owner` and I would rather everything be left on `owner` to keep it all consistent.

Reply:
Echoing what san said about edge cases where the usr doesn't match the owner, especially because of actions like killing a mob can undo the polymorph. It'd be goofy as hell if a player mob spilled out all their organs and inventory on the floor because of it.

Reply:
> Echoing what san said about cases where the usr doesn't match the owner, especially because of actions like killing the poly mob can undo the polymorph. It'd be goofy as hell if a different player mob spilled out all their organs and inventory on the floor because of it.

A certified goon foodtray moment. But let me try it one more time and hope to god it will work because last time it did runtime

Reply:
Hey, why don't we just add a `mob/shapeshifted` arg to this proc and ensure that whatever calls it makes use of that?

Reply:
All good!

The thing is, that should be handled by the `owner` var of the status effect.

But for some reason when I use owner it runtimes. I did get my langserver recently working though so I'll try to properly debug it now